src/hydra/dataProvider.ts

```diff
@@ -354,52 +391,63 @@ function dataProvider(
       case CREATE:
         return transformReactAdminDataToRequestBody(
           resource,
-          params.data,
+          (params as CreateParams).data,
           extraInformation,
         ).then((body) => ({
           options: {
             body,
             method: 'POST',
           },
-          url: collectionUrl,
+          url,
         }));
 
       case DELETE:
         return Promise.resolve({
           options: {
             method: 'DELETE',
           },
-          url: itemUrl,
+          url,
         });
 
       case GET_LIST:
       case GET_MANY_REFERENCE: {
         const {
           pagination: { page, perPage },
           sort: { field, order },
-        } = params;
+          filter,
+        } = params as GetListParams | GetManyReferenceParams;
 
         if (order && field) {
-          collectionUrl.searchParams.set(`order[${field}]`, order);
+          url.searchParams.set(`order[${field}]`, order);
         }
-        if (page) collectionUrl.searchParams.set('page', page);
-        if (perPage) collectionUrl.searchParams.set('itemsPerPage', perPage);
-        if (params.filter) {
-          const buildFilterParams = (key, nestedFilter, rootKey) => {
+        if (page) url.searchParams.set('page', page.toString());
+        if (perPage) url.searchParams.set('itemsPerPage', perPage.toString());
+        if (filter) {
+          const buildFilterParams = (
+            key: string,
+            nestedFilter: {
+              [key: string]:
+                | string
+                | boolean
+                | number
+                | { [key: string]: string | boolean | number };
```

Comment:
this is a valid case for `any` imo

Reply:
`Record<string, unknown>`  maybe

Reply:
Filter values have only these types: it's useful to use the exact types.